dapp/src/hooks/useNameRecord.ts

```diff
@@ -42,14 +49,26 @@ export function useNameRecord(
     { price = DEFAULT_PRICE_OPTION }: UseNameRecordOptions = {},
 ) {
     const { iotaNamesClient } = useIotaNamesClient();
+    const { reservedList, blockedList } = useDenyList();
 
+    const isNameReserved = reservedList.some((label) => name.includes(label));
+    const isNameBlocked = blockedList.some((label) => name.includes(label));
     return useQuery({
-        queryKey: [...queryKey.nameRecord(name), price],
+        queryKey: [...queryKey.nameRecord(name), price, isNameReserved, isNameBlocked],
         async queryFn() {
             const validationError = validateIotaName(name);
             if (validationError) {
                 throw new Error(validationError);
             }
+
+            if (isNameReserved || isNameBlocked) {
```

Comment:
This is not good because the dapp will break if someone has the "reserved name" in his "My Names" page, as this with early return a `null` data